arrows/pytorch/track.py

```diff
@@ -0,0 +1,270 @@
+        self._sys_frame_id = val
+
+    @property
+    def sys_frame_time(self):
+        return self._sys_frame_time
+
+    @sys_frame_time.setter
+    def sys_frame_time(self, val):
+        self._sys_frame_time = val
+
+class track(object):
+    def __init__(self, id):
+        self._track_id = id
+        self._track_state_list = []
+        self._max_conf = 0.0
+        self._updated_flag = False
+
+    def __len__(self):
+        return len(self._track_state_list)
+
+    def __getitem__(self, idx):
+        return self._track_state_list[idx]
+
+    def __iter__(self):
+        return iter(self._track_state_list)
+
+    @property
+    def id(self):
+        return self._track_id
+
+    @id.setter
+    def id(self, val):
+        self._track_id = val
+
+    @property
+    def updated_flag(self):
+        return self._updated_flag
+
+    @updated_flag.setter
+    def updated_flag(self, val):
+        self._updated_flag = val
+
+    @property
+    def track_state_list(self):
+        return self._track_state_list
+
+    @track_state_list.setter
+    def track_state_list(self, val):
+        self._track_state_list = val
+
+    @property
+    def max_conf(self):
+        return self._max_conf
+
+    @max_conf.setter
+    def max_conf(self, val):
+        self._max_conf = val
+
+    def append(self, new_track_state):
+        if not self._track_state_list:
+            new_track_state.motion_feature = torch.FloatTensor(2).zero_()
+        else:
+            pre_bbox_center = np.asarray(self._track_state_list[-1].bbox_center, dtype=np.float32).reshape(2)
+            cur_bbox_center = np.asarray(new_track_state.bbox_center, dtype=np.float32).reshape(2)
+            new_track_state.motion_feature = torch.from_numpy(cur_bbox_center - pre_bbox_center)
+
+        new_track_state.track_id = self._track_id
+        self._track_state_list.append(new_track_state)
+        self._max_conf = max(self._max_conf, new_track_state.conf)
+
+    def duplicate_track_state(self, timestep_len = 6):
+        #du_track = copy.deepcopy(self)
+        du_track = track(self._track_id)
+        du_track.track_state_list = list(self._track_state_list)
+        du_track.updated_flag = self._updated_flag
+        du_track.max_conf = self._max_conf
+
+        for _ in range(timestep_len - len(du_track)):
+            du_track.append(du_track[-1])
+
+        return du_track
+
+
+class track_set(object):
+    def __init__(self):
+        self._id_ts_dict = collections.OrderedDict()
+        # We implement an ordered set by mapping to None
+        self._active_id_set = collections.OrderedDict()
+
+    def __len__(self):
+        return len(self._id_ts_dict)
+
+    def __iter__(self):
+        return iter(self._id_ts_dict.values())
+
+    def iter_active(self):
+        return (self[i] for i in self._active_id_set)
+
+    def __getitem__(self, track_id):
+        try:
+            return self._id_ts_dict[track_id]
+        except KeyError:
+            raise IndexError
```

Comment:
Unless callers depend on this raising an `IndexError` instead of a `KeyError`, this could be simplified to just
```python
return self._id_ts_dict[track_id]
```